code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1030 @@
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer nilTxnServer.Close()
+
+	type args struct {
+		txnHash string
+		urls    []string
+		sf      int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *Transaction
+		wantErr bool
+	}{
+		{
+			name: "ERR",
+			args: args{
+				urls: []string{
+					"worng url",
+					errServer.URL,
+					invServer.URL,
+					nilTxnServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+			want:    &txn,
+			wantErr: false,
+		},
+		{
+			name:    "Txn_Not_Found_ERR",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetTransactionStatus(tt.args.txnHash, tt.args.urls, tt.args.sf)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetTransactionStatus() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetTransactionStatus() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMakeSCRestAPICall(t *testing.T) {
+	errEntity := mocks.Serializable{}
+	errEntity.On("Decode", mock.AnythingOfType("[]uint8")).Return(
+		func(blob []byte) error {
+			return errors.New("")
+		},
+	)
+
+	entity := mocks.Serializable{}
+	entity.On("Decode", mock.AnythingOfType("[]uint8")).Return(
+		func(blob []byte) error {
+			return nil
+		},
+	)
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+			},
+		),
+	)
+	defer server.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer errServer.Close()
+
+	type args struct {
+		scAddress    string
+		relativePath string
+		params       map[string]string
+		urls         []string
+		entity       util.Serializable
+		consensus    int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Client_ERR",
+			args: args{
+				params: map[string]string{
+					"key": "value",
+				},
+				urls: []string{
+					"wrong url",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name:    "Empty_URLs_ERR",
+			wantErr: true,
+		},
+		{
+			name: "Response_Status_Not_Ok_ERR",
+			args: args{
+				urls: []string{
+					errServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Response_Body_Decode_ERR",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				entity: &errEntity,
+			},
+			wantErr: true,
+		},
+		{
+			name: "Consensus_Success_OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				entity: &entity,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Consensus_ERR",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				entity:    &entity,
+				consensus: 200,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Closure error, missing `tt:=tt`.

Reply:
There is no race between goroutines. Test running order is determined by tests struct slice.

Reply:
Fixed.